x/lockup/keeper/migration.go

```diff
@@ -94,7 +94,7 @@ func MergeLockupsForSimilarDurations(
 				}
 				// create a normalized lock that will absorb the locks in the duration window
 				normalID = k.GetLastLockID(ctx) + 1
-				normalLock = types.NewPeriodLock(normalID, owner, normalizedDuration, time.Time{}, lock.Coins)
+				normalLock = types.NewPeriodLock(normalID, owner, lock.RewardReceiverAddress, normalizedDuration, time.Time{}, lock.Coins)
```

Comment:
This also feels like unexpected behavior. What if we have 5 locks, all with different rewardReceiverAddresses? This would set the normalized lock to whatever the final lock's rewardRecieverAddress is. Should we require that all locks have the same rewardReceiverAddress in order to merge them?

Reply:
IMO we should not be using this method anymore, since this was a method created use for migration in v6, and our codebase has changed alot since then. 

however, I do think that your case states a valid point thus added your suggestion above as a saftey measure just in case we do come across using this method

Reply:
If this method is unused, can we remove it?

Reply:
We still need the method though since we have the old upgrade handler :(